### ISSUE TITLE
模型预训练

### ISSUE THREAD

**Issue by ZF1546** [solitaire2015/Split_Merge_table_recognition#3]
真的很巧，刚给你发消息，之前的对话窗关闭了，哈哈。我这边已经训练了，但是效果不理想，如果方便的话，可以共享你们的最终模型吗，我想在你们的最终模型基础上微调训练我的数据可能会更好一点，谢谢！

**Comment by solitaire2015** [comment]
@ZF1546 是我们的最终模型了。效果不好具体的表现是什么呢？

**Comment by ZF1546** [comment]
@solitaire2015 你好，效果不好主要出现在线条检测出现多检和少检的情况。下面是原图和split测试的结果。
![0](https://user-images.githubusercontent.com/45488462/74808569-1a1a1100-5326-11ea-8960-d92f38b7958d.JPG)
![1](https://user-images.githubusercontent.com/45488462/74808580-22724c00-5326-11ea-839c-079431f8a0ac.jpg)
![2](https://user-images.githubusercontent.com/45488462/74808591-27cf9680-5326-11ea-9fd1-fd967cb769a9.jpg)
![3](https://user-images.githubusercontent.com/45488462/74808598-2bfbb400-5326-11ea-9ed3-c6bc736b470b.jpg)

**Comment by solitaire2015** [comment]
https://github.com/solitaire2015/Split_Merge_table_recognition/blob/a5c270cd0c3dc5cc52a8069356219764e587be37/loss/loss.py#L10
@ZF1546  在这里我使用BCE loss的时候加了一个正负样本的权重，你可以把这个调小一点，另外，你有没有观察模型在训练集和测试集的效果上的差别，是否有过拟合。
使用Split&Merge的时候，输入是切出来的表格块，而不是原来的整图，之前还是有一步table_detection的，我的数据处理脚本也是先切出表格的块，再用切出来的块训练Split 和 Merge模型的。
你可以尝试一下。

**Comment by ZF1546** [comment]
@solitaire2015 你好，训练集我也测试了一下，结果差不多，我一会按你说的调整一下，明天再训练一下试试。
刚才我用你的模型进行了测试，也出现了同样的效果问题，我只是对你给的测试脚本稍微修改了一下，不知道是不是测试脚本的问题。
![原图](https://user-images.githubusercontent.com/45488462/74811295-230de100-532c-11ea-9baf-b66b64a6fa23.png)
![1](https://user-images.githubusercontent.com/45488462/74811304-27d29500-532c-11ea-8ac8-dd0b05fa6d6f.jpg)
![2](https://user-images.githubusercontent.com/45488462/74811310-2acd8580-532c-11ea-8073-9afe021675d9.jpg)
![测试脚本](https://user-images.githubusercontent.com/45488462/74811332-3b7dfb80-532c-11ea-83a7-5f28d1adeaa9.png)

**Comment by ZF1546** [comment]
`import os
from dataset.dataset import ImageDataset
from modules.split_modules import SplitModel
import json
from PIL import Image
import torch
from torchvision import transforms
from torchsummary import summary
import numpy as np
import cv2
import json
import matplotlib.pyplot as plt
import numpy

os.environ['CUDA_VISIBLE_DEVICES'] = '0'

img = Image.open("./test/13.png").convert('RGB')
x, y = img.size[0:2]
img = img.resize((int(x / 2), int(y / 2)))
img = np.array(img)
img_array = np.array(img) / 255.
img_tensor = torch.from_numpy(img_array).type(torch.FloatTensor)
img_tensor=img_tensor.permute(2,0,1)
print(img_tensor.shape)
img_tensor=img_tensor.reshape(-1,img_tensor.shape[0],img_tensor.shape[1],img_tensor.shape[2])
# load model
net = SplitModel(3)
net = torch.nn.DataParallel(net).cuda()

# load saved checkpoint
net.load_state_dict(torch.load('./split/split_models/split_best.pth'))

net.eval()

r,c = net(img_tensor)
print(type(r))

r = r[-1]>0.5
c = c[-1]>0.5
c = c.cpu().detach().numpy()
r = r.cpu().detach().numpy()

print(r)
r_im = r.reshape((-1,1))*np.ones((r.shape[0],c.shape[0]))
c_im = c.reshape((1,-1))*np.ones((r.shape[0],c.shape[0]))
print(r_im)

im = cv2.bitwise_or(r_im,c_im)

cv2.imwrite('1.jpg', c_im*255)
cv2.imwrite('2.jpg', r_im*255)
cv2.imwrite('3.jpg',im*255)

`

**Comment by solitaire2015** [comment]
@ZF1546 看起来没啥问题，你先试试，我也再想想有没有要修改的点。

**Comment by ZF1546** [comment]
@solitaire2015 好的，麻烦你啦，谢谢！
